src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -105,16 +158,58 @@ public static ISymbol GetFirstSymbolForDeclarationId(string id, Compilation comp
         /// <summary>
         /// Gets the symbols that match the reference id string.
         /// </summary>
-        public static IEnumerable<ISymbol> GetSymbolsForReferenceId(string id, Compilation compilation)
+        public static ImmutableArray<ISymbol> GetSymbolsForReferenceId(string id, Compilation compilation)
         {
+            if (id == null)
+            {
+                throw new ArgumentNullException(nameof(id));
+            }
+
+            if (compilation == null)
+            {
+                throw new ArgumentNullException(nameof(compilation));
+            }
+
+            var results = s_symbolListPool.Allocate();
+            try
+            {
+                TryGetSymbolsForReferenceId(id, compilation, results);
+                return results.ToImmutableArray();
+            }
+            finally
+            {
+                s_symbolListPool.ClearAndFree(results);
+            }
+        }
+
+        /// <summary>
+        /// Try to get all symbols that match the reference id string.
+        /// Returns true if atleast one symbol matches.
+        /// </summary>
+        /// <remarks>This API is available for possible of reuse of results lists.</remarks>
+        public static bool TryGetSymbolsForReferenceId(string id, Compilation compilation, List<ISymbol> results)
```

Comment:
:bulb: My suggestion here is the same as above (consider making this method internal).